corehq/apps/app_manager/tasks.py

```diff
@@ -91,3 +93,54 @@ def update_linked_app_and_notify_task(domain, app_id, master_app_id, user_id, em
 def load_appcues_template_app(domain, username, app_slug):
     from corehq.apps.app_manager.views.apps import load_app_from_slug
     load_app_from_slug(domain, username, app_slug)
+
+
+@task(queue='background_queue', ignore_result=True)
+def analyse_new_app_build(domain, new_build_id):
+    new_build = get_app(domain, new_build_id)
+
+    check_for_custom_callouts(new_build)
+    check_build_dependencies(new_build)
+
+
+def check_for_custom_callouts(new_build):
+    from corehq.apps.app_manager.util import app_callout_templates_ids
+    template_ids = app_callout_templates_ids()
+
+    def app_has_custom_intents():
+        return any(
+            any(set(form.wrapped_xform().odk_intents) - template_ids)
+            for form in new_build.get_forms()
+        )
+
+    if app_has_custom_intents():
+        metrics_counter(
+            'commcare.app_build.custom_app_callout',
```

Comment:
how does this work if an app with custom app callout, keeps making new versions.
This metric keeps getting notified for the same app?

Reply:
Correct.

Reply:
Essentially, AEs are interested to know how many apps have custom intents. As it's now Datadog will show the app_id, domain and a number - the number of builds with custom intents (not retrospectively, but since this has been deployed). I guess they don't have to see the number of builds with custom intents and we could only show them if an app have custom intents as per latest build, which would be totally fine.

If we decide the latter this counter could become a gauge.